helix-view/src/editor.rs

```diff
@@ -1095,38 +1114,21 @@ impl Editor {
         self._refresh();
     }
 
-    pub async fn close_document(
-        &mut self,
-        doc_id: DocumentId,
-        force: bool,
-    ) -> Result<(), CloseError> {
+    pub fn close_document(&mut self, doc_id: DocumentId, force: bool) -> Result<(), CloseError> {
         let doc = match self.documents.get_mut(&doc_id) {
             Some(doc) => doc,
             None => return Err(CloseError::DoesNotExist),
         };
-
-        // flush out any pending writes first to clear the modified status
-        if let Some(Err(err)) = doc.try_flush_saves().await {
```

Comment:
Do we not need this any more? I was hitting an issue here before where the document would not close during `write-quit` because the write has to have a chance to be driven to completion and the modified status reset so that the following check didn't fail.

Reply:
After some testing locally this doesn't appear to be a problem any more. Not sure what was causing it any more. But it's clear now from where it's used that the writes are all flushed before this runs, so it makes to me now.

Reply:
Yeah I think that's why I removed this bit 👍🏻